src/KeyHelper.sol

```diff
@@ -0,0 +1,90 @@
+pragma solidity ^0.8.13;
+
+import "./KeyManager.sol";
+
+contract AuthHelper {
+    KeyManager_v0 private keymgr;
+    address owner;
+
+    constructor(KeyManager_v0 _keymgr) {
+        keymgr = _keymgr; // Note that we only derive privkey, so we don't require derivekey be called with the address of this (private) contract. The key manager has to be bootstrapped.
+        owner = msg.sender; // Can we really trust the sender though? The kettle could be spoofing it
+    }
+
+    function hashSecret() public returns (bytes32) {
+        // We deploy a different contract as the derivePriv will be different in this scenario!
+        require(msg.sender == owner);
```

Comment:
wouldn't this restrict it to only one owner?
Would that case an issue for the case of several kettles onboarded and share the keymgr deriveKey?

Reply:
I think in this case the sender in this case will be the parent (onchain) contract, which is available to everyone